packages/framework/src/hooks/useWidgetId.ts

```diff
@@ -26,12 +27,12 @@ export const useWidgetId = (
       );
     }
     return generateRandomString(6);
-  }, [customScope, id]);
+  }, [customScope, id, testId]);
 
   const rootRef = useCallback(
     (node: never) => {
       if (node && "setAttribute" in node) {
-        (node as HTMLElement).setAttribute("data-testid", resolvedWidgetId);
```

Comment:
keep this the same

Reply:
should we fallback to `resolvedWidgetId` if neither id nor the testId is present?